### PR TITLE
Added typescript language support for jsx fragment node

### LINKED ISSUE #1450: React Fragment syntax unsupported
It seems that cursorless doesn't identify React fragment syntax, ie `<></>`. Ideally it would be treated like any other xml element. It's just syntactic sugar for `<React.Fragment></React.Fragment>`.

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
Fixes #1450

## Checklist

- [x] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [ ] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [x] I have not broken the cheatsheet

### CODE REVIEW

File: packages/cursorless-engine/src/languages/typescript.ts [hunk 1]
```diff
@@ -70,8 +71,34 @@ const STATEMENT_TYPES = [
   "with_statement",
 ];
 
-const getStartTag = patternMatcher("jsx_element.jsx_opening_element!");
-const getEndTag = patternMatcher("jsx_element.jsx_closing_element!");
+function jsxTag(isStartTag: boolean): NodeMatcher {
```
Comment (pokey) [inline packages/cursorless-engine/src/languages/typescript.ts:74]: I'd rename to `getJsxFragmentTag`

**Review by pokey — APPROVED** [review]
He's back! 😊

Looks good to me with minor comment. Nice tests

File: packages/cursorless-engine/src/languages/typescript.ts [hunk 2]
```diff
@@ -70,8 +71,34 @@ const STATEMENT_TYPES = [
   "with_statement",
 ];
 
-const getStartTag = patternMatcher("jsx_element.jsx_opening_element!");
-const getEndTag = patternMatcher("jsx_element.jsx_closing_element!");
+function getJsxFragmentTag(isStartTag: boolean): NodeMatcher {
```
Comment (pokey) [inline packages/cursorless-engine/src/languages/typescript.ts:74]:
```suggestion
/** Handles jsx fragment start or end tag, eg the `<>` in `<>foo</>` **/
function getJsxFragmentTag(isStartTag: boolean): NodeMatcher {
```

**Review by pokey — COMMENTED** [review]
Nice. One more thought: do we want to support "name tags"? Might be useful for turning fragment into proper element

**Comment by AndreasArvidsson** [comment]
> Nice. One more thought: do we want to support "name tags"? Might be useful for turning fragment into proper element

I think we wait with that one until we have the new syntax?

**Comment by pokey** [comment]
sure if you don't think it's a big deal to be missing that functionality / consistency.  I don't use jsx too frequently so not a huge deal

**Review by pokey — APPROVED** [review]

**Comment by AndreasArvidsson** [comment]
> sure if you don't think it's a big deal to be missing that functionality / consistency. I don't use jsx too frequently so not a huge deal

I don't in this case since there actually isn't a name in the tag. We don't support take type in functions without a type either. Definitely something that we should support in the future, but for now I think this is fine.

**Pull request merged**